lib/msal-node/test/client/ManagedIdentitySources/Imds.spec.ts

```diff
@@ -61,6 +61,7 @@ describe("Acquires a token successfully via an IMDS Managed Identity", () => {
     afterEach(() => {
         delete ManagedIdentityClient["identitySource"];
         delete ManagedIdentityApplication["nodeStorage"];
+        jest.restoreAllMocks();
```

Comment:
Why did you make these changes to this file? Do all the unit tests still run after this change?

Reply:
I realized while working on this PR that since this line wasn't in the afterEach, if a test in this file failed, it would show dozens of failures as a result. This change isn't directly related to this PR, I just thought I should fix it while I'm at it

Reply:
Sorry, I don't quite understand. Did any tests fail in this file that drew your attention to the file?

> if a test in this file failed, it would show dozens of failures as a result

I never observed this while developing managed identity and writing these tests.

Anyways, we already have a `jest.restoreAllMocks();` at the end of every test where a spy is being created/used, so I'm not sure what adding this here would do.

Additionally, the rest of the managed identity spec files use same logic for `jest.restoreAllMocks();` placement as well.

Reply:
Mocks should always be cleared in the before or after hooks, never at the end of a test. Reason is that if there is any failure in the test your restore line will not be executed and may cause other tests to fail as a result of mocks that are still registered. This can make things really hard to debug as it's not immediately clear that the problem isn't with the test or the code but rather improperly cleaned up mocks in an unrelated test.

Reply:
Ah, thanks for the explanation Thomas! Essentially what Shyla said :)

Are you both ok with making this a separate PR then and including all the managed identity spec files that use `jest.restoreAllMocks();`?